auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import exceptions, models, fields, api, modules, _
+from odoo.addons.auditlog.models.rule import FIELDS_BLACKLIST
+from odoo import SUPERUSER_ID
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    auditlog_line_access_rule_ids = fields.One2many(
+        "auditlog.line.access.rule", "auditlog_rule_id", ondelete="cascade"
+    )
+    server_action_id = fields.Many2one('ir.actions.server', "Server Action")
+
+
+    def sudo(self, user=SUPERUSER_ID):
+       return super(
+            AuditlogRule, self.with_context(real_user=self.env.context.get('uid'))
+            ).sudo(user=user)
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.multi
+    def unlink(self):
+        lines = self.mapped("auditlog_line_access_rule_ids")
+        res = super(AuditlogRule, self).unlink()
+        if res:
+            lines.unlink()
+        return res
```

Comment:
does this still lead to trouble? and shouldnt we do this on "Unsubscribe" instead?

Reply:
logically, yes. removing.